src/acceptance-test/java/net/consensys/orion/acceptance/EthClientStub.java

```diff
@@ -172,6 +173,20 @@ ReceiveResponse receivePrivacy(final String digest, final String publicKey) {
     return Optional.ofNullable(keyFuture.join());
   }
 
+  public Optional<PrivacyGroup> retrievePrivacyGroup(final String privacyGroupId) {
+    final RetrievePrivacyGroupRequest getGroupRequest = new RetrievePrivacyGroupRequest(privacyGroupId);
+    final CompletableFuture<PrivacyGroup> keyFuture = new CompletableFuture<>();
+    httpClient.post(clientPort, "localhost", "/retrievePrivacyGroup").handler(resp -> {
+      if (resp.statusCode() == 200) {
+        resp.bodyHandler(body -> keyFuture.complete(deserialize(body, PrivacyGroup.class)));
+      } else {
+        keyFuture.complete(null);
+      }
+    }).exceptionHandler(keyFuture::completeExceptionally).putHeader("Content-Type", "application/json").end(
+        Buffer.buffer(Serializer.serialize(JSON, getGroupRequest)));
+    return Optional.ofNullable(keyFuture.join());
```

Comment:
Can this still throw an exception?

Reply:
No it won't ever thrown an exception as the keyFuture is always completed successfully.